canopen/objectdictionary/__init__.py

```diff
@@ -116,13 +116,14 @@ class Record(MutableMapping):
     #: Description for the whole record
     description = ""
 
-    def __init__(self, name, index):
+    def __init__(self, name, index, StorageLocation):
```

Comment:
The way I added extra information to the Variable object is to write those attributes after it has been created. This could definitely be debated but it keeps constructor clean and backwards compatible. So I would prefer if you removed it from here and just write the value later.

Reply:
How can i add extra information later for an `Array` or `Record` ? Do i need to make a funciton my own or are there functions to do that inside the code?

Reply:
```python
arr = objectdictionary.Array(name, index)
arr.storage_location = storage_location
```

Similar to how it's done on Variable.